inference-engine/src/transformations/src/transformations/smart_reshape/mimic_set_batch_size.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <transformations/smart_reshape/mimic_set_batch_size.hpp>
+
+#include <numeric>
+
+#include <ngraph/ngraph.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/pattern/matcher.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+ngraph::pass::MimicSetBatchSize::MimicSetBatchSize() {
+    auto reshape_label = ngraph::pattern::wrap_type<opset5::Reshape>({pattern::any_input(), ngraph::pattern::wrap_type<opset5::Constant>()});
+
+    matcher_pass_callback callback = [=](pattern::Matcher &m) -> bool {
+        const auto &pattern_to_output = m.get_pattern_value_map();
+
+        const auto & reshape = std::dynamic_pointer_cast<opset5::Reshape>(pattern_to_output.at(reshape_label).get_node_shared_ptr());
```

Comment:
You can just use root instead of value map
```suggestion
        const auto & reshape = std::dynamic_pointer_cast<opset5::Reshape>(m.get_match_root());
```

Reply:
And BTW you don't use any specific methods of Reshape class so you don't have to cast it.